// @lambda-platform/datagrid/src/utils/data.js
import {localeText} from "./agMn.js";

export const data = (vm) => {
    return {
        isLoading: true,
        identity: "id",
        overlayLoadingTemplate:
            '<span class="ag-overlay-loading-center">Түр хүлээнэ үү...</span>',
        init: false,
        model: null,
        template: 2,
        theme: 'normal',
        isClient: false,
        isPrint: false,
        isNumbered: false,
        printSize: 'A4',
        sideBar: false,
        header: null,
        tableWidth: null,
        fullWidth: false,
        fullText: false,
        editableAction: null,
        editType: null,
        editCellValue: null,
        editRowValue: null,
        postModels: [],
        updateModels: [],
        cellStatus: '',
        singleClickEdit: true,
        flashChanges: false,
        editableShouldSubmit: false,
        gridOptions: {
            debug: false,
            floatingFilter: false,
            suppressMultiSort: true,
            suppressRowClickSelection: true,
            allowContextMenuWithControlKey: true,
            animateRows: false,
            localeText: localeText,
            suppressCellSelection: true,
            groupIncludeFooter: false,
            groupIncludeTotalFooter: false,
            suppressNoRowsOverlay: true,
            // enableCharts: true,
            // enableRangeSelection: true
        },
        info: {
            total: 0,
            totalPage: 0
        },
        query: {
            src: "",
            paginate: 1000,
            currentPage: vm.$route.query.dp ? parseInt(vm.$route.query.dp) : 1,
            sort: "",
            order: ""
        },
        gridActions: [],
        hasContextMenu: false,
        actionPosition: 0,
        schema: [],
        formula: [],
        aggregations: {
            columnAggregations: [],
            columnAggregationsFormula: [],
            data: [],
            loading: true,
            forumlaResult: ""
        },
        columns: null,
        data: [],
        filterModel: {},
        searchModel: null,
        isStatic: false,
        //for header filter
        selectionMethod: 'multiple',
        selected: [],
        autoSelect: false,
        autoSelectModel: null,
        deleteModal: false,
        delLoading: false,
        currentRow: {
            rowId: null,
            rowIndex: null
        },
        frameworkComponents: null,
        changedRowsData: [],
        colMenu: false,
        colFilterButton: true,
        showGrid: false,
        saveFilter: false,
        selectInputModels: [],
    }
}


export const builderData = (vm) => {
    return {
        loading: true,
        scrollOptions: {
            height: '100%',
            size: 7,
            alwaysVisible: true,
            wheelStep: 5,
            color: '#2C3A47'
        },
        loadingSubmit: false,
        isModelSelected: false,
        gridName: null,
        datagrid: {
            model: null,
            isView: false,
            identity: null,
            actions: ["e", "d"],
            actionPosition: 0,
            isContextMenu: false,
            staticWidth: false,
            fullWidth: true,
            hasCheckbox: false,
            isClient: false,
            width: 0,
            sort: null,
            sortOrder: "desc",
            softDelete: false,
            paging: 50,
            template: 0,
            schema: [],
            filter: [],
            formula: [],
            condition: null,
            columnAggregations: [],
            columnAggregationsFormula: [],
            header: {
                render: false,
                preview: false,
                structure: []
            },
            triggers: {
                namespace: '',
                beforeFetch: '',
                afterFetch: '',
                beforeDelete: '',
                afterDelete: '',
                beforePrint: '',
            },
            theme: 'normal',
            fullText: false,
            editableAction: null,
            editFullRow: false,
            editableShouldSubmit: false,
            singleClickEdit: true,
            flashChanges: false,
            colMenu: false,
            colFilterButton: true,
            showGrid: false,
            saveFilter: false
        },
        tableList: window.init.dbSchema.tableList,
        viewList: window.init.dbSchema.viewList,
        fieldList: [],
        gridThemes: ['normal', 'mini'],
        //Formula
        formulaForm: {
            model: "",
            template: ""
        },

        formulaRule: {
            model: [
                {
                    required: true,
                    message: "Үр дүн хадгалах талбар сонгоно уу",
                    trigger: "blur"
                }
            ],
            template: [
                {
                    required: true,
                    message: "Томъёогоо оруулна уу",
                    trigger: "blur"
                }
            ]
        },

        formulaColumns: [
            {
                title: "Томъёо",
                key: "template",
                minWidth: 200
            },
            {
                title: "Талбар",
                key: "model",
                minWidth: 150
            },
            {
                title: "Устгах",
                key: "action",
                width: 80,
                align: "center",
                render: (h, params) => {
                    return h("div", [
                        h("Button", {
                            props: {
                                type: "ghost",
                                shape: "circle",
                                icon: "android-close",
                                size: "small"
                            },
                            on: {
                                click: () => {
                                    vm.removeFormula(params.index);
                                }
                            }
                        })
                    ]);
                }
            }
        ],

        //Column Aggrigation
        aggregations: [
            "SUM",
            "COUNT",
            "MIN",
            "MAX",
            "AVG",
            "CountDistinct",
            "AvgDistinct",
            "SumDistinct"
        ],

        columnAggregationForm: {
            column: "",
            aggregation: "",
            symbol: ""
        },

        columnAggregationRule: {
            column: [
                {
                    required: true,
                    message: "Багана сонгоно уу",
                    trigger: "blur"
                }
            ],
            aggregation: [
                {
                    required: true,
                    message: "Нэгтгэл сонгоно уу",
                    trigger: "blur"
                }
            ],
            symbol: [
                {
                    required: false
                }
            ]
        },

        columnAggregationColumns: [
            {
                title: "Багана",
                key: "column",
                minWidth: 120
            },
            {
                title: "Нэгтгэл төрөл",
                key: "aggregation",
                minWidth: 150
            },
            {
                title: "Тэмдэг",
                key: "symbol",
                minWidth: 100
            },
            {
                title: "Устгах",
                key: "action",
                width: 100,
                align: "center",
                render: (h, params) => {
                    return h("div", [
                        h(
                            "Button",
                            {
                                props: {
                                    type: "ghost",
                                    shape: "circle",
                                    icon: "android-close",
                                    size: "small"
                                },
                                on: {
                                    click: () => {
                                        vm.removeColumnAggergation(
                                            params.index
                                        );
                                    }
                                }
                            },
                            "Устгах"
                        )
                    ]);
                }
            }
        ],

        //Column Aggrigation's Formula
        columnAggregationsFormulaForm: {
            title: "",
            template: "",
            symbol: ""
        },

        columnAggregationsFormulaRule: {
            template: [
                {
                    required: true,
                    message: "Томъёогоо оруулна уу",
                    trigger: "blur"
                }
            ],
            symbol: [
                {
                    required: false
                }
            ],
            title: [
                {
                    required: false
                }
            ]
        },

        columnAggregationsFormulaColumns: [
            {
                title: "Нэр",
                key: "title",
                minWidth: 100
            },
            {
                title: "Томъёо",
                key: "template",
                minWidth: 300
            },
            {
                title: "Тэмдэг",
                key: "symbol",
                minWidth: 200
            },
            {
                title: "Устгах",
                key: "action",
                width: 100,
                align: "center",
                render: (h, params) => {
                    return h("div", [
                        h(
                            "Button",
                            {
                                props: {
                                    type: "ghost",
                                    shape: "circle",
                                    icon: "android-close",
                                    size: "small"
                                },
                                on: {
                                    click: () => {
                                        vm.removeColumnAggregationsFormula(
                                            params.index
                                        );
                                    }
                                }
                            },
                            "Устгах"
                        )
                    ]);
                }
            }
        ]
    }
};
