// @lambda-platform/puzzle/src/router.js
import Router from 'vue-router'

Vue.use(Router)

let base = '/';

if (location.pathname && location.pathname != '/') {
    base = location.pathname.split('/').slice(0, -1).join('/');
}

let routes = [{
    path: '/',
    redirect: '/moqup'
},
    {
        path: '/moqup',
        component: require('./pages/moqup/views/index.vue').default,
        children: [{
            name: 'moqup.list',
            path: '',
            component: require('./pages/moqup/views/list.vue').default,
        },
            {
                name: 'moqup.builder',
                path: 'builder/:id?',
                component: require('./pages/moqup/views/builder.vue').default,
            },
            {
                name: 'moqup.preview',
                path: 'preview/:id',
                component: require('./pages/moqup/views/preview.vue').default,
            },
        ]
    },
    {
        path: '/graphql',
        component: require('./pages/graphql/views/index.vue').default,
        children: [{
            name: 'graphql.list',
            path: '',
            component: require('./pages/graphql/views/list.vue').default,
        },
            {
                name: 'graphql.builder',
                path: 'builder/:id?',
                component: require('./pages/graphql/views/builder.vue').default,
            },
        ]
    },
    {
        path: '/chart',
        component: require('./pages/chart/views/index.vue').default,
        children: [{
            name: 'chart.list',
            path: '',
            component: require('./pages/chart/views/list.vue').default,
        },
            {
                name: 'chart.builder',
                path: 'builder/:id?',
                component: require('./pages/chart/views/builder.vue').default,
            },
            {
                name: 'chart.preview',
                path: 'preview/:id',
                component: require('./pages/chart/views/preview.vue').default,
            },
        ]
    },
    {
        path: '/analytic',
        component: require('./pages/analytic/views/index.vue').default
    },
    {
        path: '/notification',
        component: require('./pages/notification/views/index.vue').default
    },


    {
        path: '/datasource',
        component: require('./pages/datasource/views/index.vue').default,
        children: [{
            name: 'data-source.list',
            path: '',
            component: require('./pages/datasource/views/list.vue').default,
        },
            {
                name: 'data-source.builder',
                path: 'builder/:id?',
                component: require('./pages/datasource/views/builder.vue').default,
            }
        ]
    },
    {
        path: '/form',
        component:  require('./pages/form/views/index.vue').default,
        children: [{
            name: 'form.list',
            path: '',
            component:  require('./pages/form/views/list.vue').default,
        },
            {
                name: 'form.builder',
                path: 'builder/:id?',
                component:  require('./pages/form/views/builder.vue').default,
            },
            {
                name: 'form.preview',
                path: 'preview/:id',
                component:  require('./pages/form/views/preview.vue').default,
            },
        ]
    },
    {
        path: '/grid',
        component: require('./pages/grid/views/index.vue').default,
        children: [{
            name: 'grid.list',
            path: '',
            component: require('./pages/grid/views/list.vue').default,
        },
            {
                name: 'grid.builder',
                path: 'builder/:id?',
                component: require('./pages/grid/views/builder.vue').default,
            },
            {
                name: 'grid.preview',
                path: 'preview/:id',
                component: require('./pages/grid/views/preview.vue').default,
            },
        ]
    },

    {
        path: '/embed/form/:id',
        component: require('./pages/embed/form.vue').default,
    },

    {
        path: '/embed/grid/:id',
        component: require('./pages/embed/grid.vue').default,
    },

    {
        path: '/role',
        component: require('./pages/role/views/index.vue').default
    },

    {
        path: '/crud',
        component: require('./pages/crud/views/index.vue').default
    },

    {
        path: '/menu',
        component: require('./pages/menu/views/index.vue').default
    },

    {
        path: '/module/:module',
        component: require('./pages/render/views/module.vue').default
    },

    {
        name: 'Error',
        path: '/*',
        component: require('./pages/error/views/404.vue').default
    }
]

export default new Router({
    base,
    routes,
    history: true,
    linkActiveClass: 'active'
})
